include/openssl/kdf.h

```diff
@@ -10,10 +10,48 @@
 #ifndef HEADER_KDF_H
 # define HEADER_KDF_H
 
+# include <openssl/ossl_typ.h>
 # include <openssl/kdferr.h>
-#ifdef __cplusplus
+# ifdef __cplusplus
 extern "C" {
-#endif
+# endif
+
+# define EVP_KDF_PBKDF2     NID_id_pbkdf2
+# define EVP_KDF_SCRYPT     NID_id_scrypt
+# define EVP_KDF_TLS1_PRF   NID_tls1_prf
+# define EVP_KDF_HKDF       NID_hkdf
+
+EVP_KDF_CTX *EVP_KDF_CTX_new_id(int id);
+void EVP_KDF_CTX_free(EVP_KDF_CTX *ctx);
+
+void EVP_KDF_reset(EVP_KDF_CTX *ctx);
+int EVP_KDF_ctrl(EVP_KDF_CTX *ctx, int cmd, ...);
+int EVP_KDF_vctrl(EVP_KDF_CTX *ctx, int cmd, va_list args);
+int EVP_KDF_ctrl_str(EVP_KDF_CTX *ctx, const char *type, const char *value);
+size_t EVP_KDF_size(EVP_KDF_CTX *ctx);
+int EVP_KDF_derive(EVP_KDF_CTX *ctx, unsigned char *key, size_t keylen);
+
+
+# define EVP_KDF_CTRL_SET_PASS          0x01 /* unsigned char *, size_t */
+# define EVP_KDF_CTRL_SET_SALT          0x02 /* unsigned char *, size_t */
+# define EVP_KDF_CTRL_SET_ITER          0x03 /* int */
+# define EVP_KDF_CTRL_SET_MD            0x04 /* EVP_MD * */
+# define EVP_KDF_CTRL_SET_KEY           0x05 /* unsigned char *, size_t */
+# define EVP_KDF_CTRL_SET_MAXMEM_BYTES  0x06 /* uint64_t */
+# define EVP_KDF_CTRL_SET_TLS_SECRET    0x07 /* unsigned char *, size_t */
```

Comment:
Could this be renamed to remove the `TLS_` prefix? likewise for seed below.
When using the string ctrl form they are just `"secret"` and `"seed"`.